cli/lib/webpack/service.js

```diff
@@ -0,0 +1,239 @@
+const crypto = require('crypto');
+const EventEmitter = require('events');
+const util = require('util');
+
+const { ProgressBarReporter, SimpleReporter } = require('./reporter');
+const appcd = require('./appcd');
+
+const STATE_READY = 'ready';
+const STATE_BUILDING = 'building';
+const STATE_STOPPED = 'stopped';
+const STATE_ERROR = 'error';
+
+class WebpackService extends EventEmitter {
+	constructor({ client, builder, cli, logger, projectType }) {
+		super();
+
+		appcd.initialize(client);
+		this.logger = logger;
+		this.builder = builder;
+		const tiapp = cli.tiapp;
+		this.type = projectType;
+		this.projectData = {
+			path: cli.argv['project-dir'],
+			name: tiapp.name,
+		};
+		this.loadBuildSettings(builder, cli);
+		this.shouldWatch = builder.deployType !== 'production' && !builder.buildOnly;
+
+		this.forceRebuild = cli.argv.force;
+		this.jobIdentifier = this.generateJobIdentifier();
+		this.pluginVersion = process.env.APPCD_WEBPACK_VERSION || 'latest';
+		this.basePath = `/webpack/${this.pluginVersion}`;
+		this.webpackStatus = { state: 'unknown' };
+		this.reporter = (!cli.argv.quiet && cli.argv['progress-bars'])
+			? new ProgressBarReporter()
+			: new SimpleReporter(logger);
+		this.webUiUrl = null;
+		this.updateTimeout = null;
+	}
+
+	loadBuildSettings(builder, cli) {
+		const platform = cli.argv.platform;
+		const buildSettings = {
+			platform,
+			deployType: builder.deployType,
+			target: cli.argv.target,
+			sdk: {
+				path: cli.sdk.path,
+				version: cli.sdk.name,
+				gitHash: cli.sdk.manifest.githash,
+				buildDate: cli.sdk.manifest.timestamp
+			}
+		};
+
+		if (platform === 'ios') {
+			buildSettings.ios = {
+				deviceFamily: builder.deviceFamily
+			};
+		}
+
+		this.buildSettings = buildSettings;
+	}
+
+	async ensureDaemonIsRunning() {
+		try {
+			await appcd.connect();
+		} catch (e) {
+			if (e.code === 'ECONNREFUSED') {
+				this.logger.info('Starting Appcelerator Daemon (subsequent builds will be faster) ...');
+				await appcd.connect({ startDaemon: true });
+			} else {
+				throw e;
+			}
+		}
+
+		const { host, port } = appcd.client;
+		this.webUiUrl = `http://${host}:${port}${this.basePath}/web`;
+		this.logger.info(`🖥  Web UI ready on ${this.webUiUrl.cyan}`);
```

Comment:
FYI: One of the topics the CLI style guide team has on the table is the use of emojis. I believe at this point, they are discouraged, but likely to not be permitted.

Reply:
Buuhhhh, but i wanted to be one of cool kids! JK, that's the only emoji i use so not that big of a deal to remove it.